refactor events utils in e2e

danielqsj opened the pull request:
<!--  Thanks for sending a pull request!  Here are some tips for you:

1. If this is your first time, please read our contributor guidelines: https://git.k8s.io/community/contributors/guide#your-first-contribution and developer guide https://git.k8s.io/community/contributors/devel/development.md#development-guide
2. Please label this pull request according to what type of issue you are addressing, especially if this is a release targeted pull request. For reference on required PR/issue labels, read here:
https://git.k8s.io/community/contributors/devel/sig-release/release.md#issuepr-kind-label
3. Ensure you have added or ran the appropriate tests for your PR: https://git.k8s.io/community/contributors/devel/sig-testing/testing.md
4. If you want *faster* PR reviews, read how: https://git.k8s.io/community/contributors/guide/pull-requests.md#best-practices-for-faster-reviews
5. If the PR is unfinished, see how to mark it: https://git.k8s.io/community/contributors/guide/pull-requests.md#marking-unfinished-pull-requests
-->

**What type of PR is this?**

/kind feature
/sig testing

**What this PR does / why we need it**:

`test/e2e/common` is a mixture of tests (`test/e2e/common/downward_api.go`) and utility code (`test/e2e/common/events.go`). It would be nice to refactor it so that the utility code can be imported without also importing the tests. Most likely a better place for the utility code is under `test/e2e/framework`.

For this PR, the specific reason was the usage of `WaitTimeoutForEvent` in a storage test, which then made it impossible to import the storage tests without those other tests. See #83609. Once `WaitTimeoutForEvent` is in an independent utility package, the copied waitTimeoutForEvent can be removed from `test/e2e/storage/testsuites/volumemode.go` again.

**Which issue(s) this PR fixes**:
<!--
*Automatically closes linked issue when PR is merged.
Usage: `Fixes #<issue number>`, or `Fixes (paste link of issue)`.
_If PR is about `failing-tests or flakes`, please post the related issues/tests in a comment and do not use `Fixes`_*
-->

part of https://github.com/kubernetes/kubernetes/issues/83937

**Special notes for your reviewer**:

/cc @pohly

**Does this PR introduce a user-facing change?**:
<!--
If no, just write "NONE" in the release-note block below.
If yes, a release note is required:
Enter your extended release note in the block below. If the PR requires additional action from users switching to the new release, include the string "action required".

For more information on release notes see: https://git.k8s.io/community/contributors/guide/release-notes.md
-->
```release-note
NONE
```

**Additional documentation e.g., KEPs (Kubernetes Enhancement Proposals), usage docs, etc.**:

<!--
This section can be blank if this pull request does not require a release note.

When adding links which point to resources within git repositories, like
KEPs or supporting documentation, please reference a specific commit and avoid
linking directly to the master branch. This ensures that links reference a
specific point in time, rather than a document that may change over time.

See here for guidance on getting permanent links to files: https://help.github.com/en/articles/getting-permanent-links-to-files

Please use the following format for linking documentation:
- [KEP]: <link>
- [Usage]: <link>
- [Other doc]: <link>
-->
```docs

```

smarterclayton committed:
Detect watch protocol errors via an e2e test for apimachinery

This e2e test reproduces #62175 and will be expanded to check for other
negotiation related errors against real servers.

smarterclayton committed:
Create a shim for Codecs that handles client duties

Clients have to renegotiate frequently, and the shape of the interface
that a client wants is slightly different than the interface on the
server. Create a new ClientNegotiator interface that represents what
the client->server code would want to use (mostly, still evolving
Encoder) and move versioning details out of RESTClient.

In the long run, we want to remove internal clients and conversions
from clients. This takes a step in that direction and also makes sure
watch negotiation is consistent with the server.

smarterclayton committed:
Remove deprecated-dynamic client

It is now unused.

johnSchnake committed:
Print progress updates to stdout and publish to URL

One common frustration of end users running the e2e suite is that
they take a significant amount of time and it is difficult to
gauge progress.

Even if tailing the logs it can be difficult to see where one
test starts and another ends or understand the if there have been
failures in the past 1h of logs.

This change adds a new custom reporter which prints summary information
as tests complete. This includes the number of tests to run and how
many have been passed/failed/skipped along with which tests have failed.

A new flag can be set which pushes these values to an endpoint. This is
intended for integration with Sonobuoy but any endpoint could consume and
surface this data to the user so they can better understand the state
of the current test run.

xing-yang committed:
VolumeSnapshot CRD v1beta1: Enable VolumeSnapshotDataSource feature gate and update e2e tests

danielqsj committed:
Fix data race in client-go UpdateTransportConfig

andrewsykim committed:
node controller: set both deprecated Beta and GA labels for Zone/Region topology

Signed-off-by: Andrew Sy Kim <kiman@vmware.com>

andrewsykim committed:
kubelet: set both deprecated Beta and GA labels for zone/region topology from the cloud provider

Signed-off-by: Andrew Sy Kim <kiman@vmware.com>

andrewsykim committed:
core/v1: update well known labels for zones/regions to topology.kubernetes.io/zone and topology.kubernetes.io/region, mark beta labels as deprecated

Signed-off-by: Andrew Sy Kim <kiman@vmware.com>

codenrhoden committed:
copy exec file to preserve history

codenrhoden committed:
move exec files back

codenrhoden committed:
Move mount/fake.go to mount/fake_mount.go

This patch moves fake.go to mount_fake.go, and follows to principle of
always returning a discrete type rather than an Interface. All callers
of "FakeMounter" are changed to instead use "NewFakeMounter()". The
FakeMounter "Log" struct member is changed to not be exported, and
instead only access through a new "GetLog()" method.

andrewsykim committed:
pkg/util/node: update GetZoneKey to check both beta and GA labels

Signed-off-by: Andrew Sy Kim <kiman@vmware.com>

andrewsykim committed:
test/e2e: check both beta and zone label for getting cluster zone

Signed-off-by: Andrew Sy Kim <kiman@vmware.com>

prameshj committed:
Support specifying a custom subnet for ILB ip

unit test for verifying custom subnet config.

simplified ip address logic.

Modified init code to always initialize eventRecorder.

removed extra import from automerge with master.

tnozicka committed:
Fix RS informer handlers and handling expectations on delete

tnozicka committed:
Update Bazel

JishanXing committed:
76093 restructure LICENSES file generation

liggitt committed:
bump github.com/json-iterator/go v1.1.8

liggitt committed:
Remove json-iterator depth patch

robscott committed:
Adding new label to indicate what is managing an EndpointSlice

This adds a new Label to EndpointSlices that will ensure that multiple
controllers or entities can manage subsets of EndpointSlices. This
label provides a way to indicate the controller or entity responsible
for managing an EndpointSlice.

To provide a seamless upgrade from the alpha release of EndpointSlices
that did not support this label, a temporary annotation has been added
on Services to indicate that this label has been initially set on
EndpointSlices. That annotation will be set automatically by the
EndpointSlice controller with this commit once appropriate Labels have
been added on the corresponding EndpointSlices.

bertinatto committed:
Promote volume limits to GA

howardjohn committed:
Add appProtocol to EndpointSlice.Port

okartau committed:
test/e2e: Reduce need to use local kubectl

Use POST method instead of running local kubectl.
Use ExecCommandInContainerWithFullOutput() instead of RunKubectl().
PodExec() takes additional framework arg, passed down in call chain.
VerifyExecInPodFail uses different error code cast as original
one causes test code Panic if used with new call method.

robscott committed:
Setting Hostname from Pods on EndpointSlice to match Endpoints behavior.

This was an oversight in the initial EndpointSlice release. This update
will ensure that Endpoints and EndpointSlices use the same logic to set
the Hostname attribute.

okartau committed:
utils.go: Changed bashExec to shExec, bash not guaranteed.

okartau committed:
fixtures.go: Removed unused parameter "client" in testVolumeContent()

mikedanese committed:
adjust token cache benchmarks to get more accurate behavior

b.N is adjusted by pkg/testing using an internal heuristic:

> The benchmark function must run the target code b.N times. During
> benchmark execution, b.N is adjusted until the benchmark function
> lasts long enough to be timed reliably.

Using b.N to seed other parameters makes the benchmark behavior
difficult to reason about. Before this change, thread count in the
CachedTokenAuthenticator benchmark is always 5000, and batch size is
almost always 1 when I run this locally. SimpleCache and StripedCache
benchmarks had similarly strange scaling.

After modifying CachedTokenAuthenticator to only adjust iterations based
on b.N, the batch chan was an point of contention and I wasn't able to
see any significant CPU consumption. This was fixed by using
ParallelBench to do the batching, rather than using a chan.

danielqsj committed:
refactor events utils in e2e

okartau committed:
Removed 2nd arg "cs" from TestVolumeClient

as we assume it is always set to f.ClientSet and we can
use its value from passed framework argument.

timothysc commented:
/assign @andrewsykim

okartau committed:
Remove argument "cs clientset.Interface" from testFlexVolume

cs is not used any more after previous change removed cs arg
from TectVolumeClient. Functions down the call path
get cs value from framework parameter.

danielqsj commented:
/cc @alejandrox1

pohly commented on test/e2e/storage/testsuites/volumemode.go:
```diff
@@ -482,16 +482,6 @@ func listPodDirectory(h utils.HostExec, path string, node *v1.Node) ([]string, e
-// waitTimeoutForEvent waits the given timeout duration for an event to occur.
-func waitTimeoutForEvent(c clientset.Interface, namespace, eventSelector, msg string, timeout time.Duration) error {
-	interval := 2 * time.Second
-	return wait.PollImmediate(interval, timeout, eventOccurred(c, namespace, eventSelector, msg))
```
I think the `eventOccurred` below can now also be removed.

danielqsj replied:
Yep, I miss it. Thanks.
PR updated, PTAL

danielqsj committed:
remove unused func

bertinatto committed:
Enable volume limits e2e tests for CSI hostpath driver

danielqsj commented:
/retest

liu-cong committed:
Add scheduler plugin execution duration metric.

Address comments

Sample metrics

Use rand.Intn and some cleanup

TheLastProject committed:
Add --certificate-key flag to token create

soltysh committed:
Set TypeSetter in get print flags, like we do everywhere else

neolit123 committed:
kubeadm: always use a short timeout for clientset creation

ToClientSet() in kubeconfig.go creates a clientset from
the passed Config object (kubeconfig). For IP addresses
that are not reachable e.g. Get() calls for ConfigMaps
can block for a few minutes with the default timeout.

Modify the timeout to a shorter value by passing an override.

andrewsykim commented on test/e2e/storage/testsuites/volumemode.go:
```diff
@@ -481,32 +480,3 @@ func listPodDirectory(h utils.HostExec, path string, node *v1.Node) ([]string, e
 	return strings.Split(out, "\n"), nil
 }
-
-// NOTE(avalluri): The below code is intentionally copied from e2e/common package.
```
Does this still apply?

pohly replied:
No, all of the copied code gets removed by this PR.

hwdef committed:
pkg/apis: fix staticcheck warning

smarterclayton committed:
test: Watch should fail immediately on negotiate errors

Instead of returning an error on the watch stream, if we can't properly
negotiate a watch serialization format we should error and return that
error to the client.

smarterclayton committed:
dynamic: The dynamic client no longer needs a special cased watch

By correctly handling content type negotiation, we can avoid the
need for a special version of watch and use the same code path as
typed clients.

smarterclayton committed:
test: Set RateLimiter via client config vs direct casting

This test has no need to perform a dereference to an implementation,
it can instead set the rate limited during initialization.

smarterclayton committed:
test: kubectl unit tests should be using codecs without conversion

Tests are also refactored to use the simpler RESTClient code path.

smarterclayton committed:
test: Exit early during resource helper test

The test will panic if it fails, and should instead check and exit
when invalid conditions are hit.

yue9944882 committed:
flowcontrol rest storage implementation

yastij committed:
remove system validators package from kubeadm and use k8s.io/system-validators instead

Signed-off-by: Yassine TIJANI <ytijani@vmware.com>

mm4tt committed:
Add envs for configuring hollow-node resource usage.

In addition clean up template variables to always use lower case in
hollow-node_template.yaml.

bertinatto committed:
Update hostpath CSI driver manifest

This is required for promoting volume limits to GA. The new version of
the driver reports the max number of volumes it supports. Such number
should be specified as a CLI argument when starting the driver.

ravisantoshgudimetla committed:
Relax namespace restriction for critical pods

liggitt committed:
Switch kubelet/aggregated API servers to use v1 tokenreviews

liggitt committed:
Switch kubelet/aggregated API servers to use v1 subjectaccessreviews

liggitt committed:
generated

deads2k committed:
add featuregate inspection as admission plugin initializer

deads2k committed:
remove global variable dep in admission

yliaog committed:
added nodeSelector: beta.kubernetes.io/os: linux

RainbowMango committed:
Migrate custom collector for kubelet

yue9944882 committed:
typo list-type s/associative/map/

deads2k committed:
allow a verifyoptionsfunc to indicate that no certpool is available

liu-cong committed:
Aggregate mulitple NodePreference custom priorities to a single score plugin.

deads2k committed:
close the kube-apiserver for taint tests

marosset committed:
Moving WindowsRunAsUserName to beta

danielqsj commented:
/cc @oomichi

RainbowMango committed:
Deal with auto-generated files.
Update bazel by hack/update-bazel.sh

andyzhangx committed:
fix race condition when attach/delete disk

fix comments

deads2k committed:
make azure fail if feature gates are not registered

k8s-ci-robot committed:
Merge pull request #84881 from oomichi/cleanup-skip.go

Remove unused skip functions

neolit123 committed:
e2e_kubeadm: fix missing suite --test* flags

For more details see 00e1ffb4e015c9a033b6c1f5446e70822af4e964.

alculquicondor committed:
Set go version in fluentd-elasticsearch addon

Signed-off-by: Aldo Culquicondor <acondor@google.com>

robscott committed:
Limiting the scope of new kubectl get e2e tests to decrease flakiness

The `kubectl get output` e2e test I'd previously added ended up being
flaky in certain e2e test scenarios. It relies on getting a list of API
resources in the cluster and running `kubectl get` calls against them.
The problem ended up being that other e2e tests could create resources
that would cause this test to fail. This change limits the scope of the
tests to not cover CRDs. This should still allow the test to catch new
Kubernetes resources with improperly configured kubectl output while
limiting the flakiness of the test.

Commit:
Fix the import aliase nodev1beta1

When running ./hack/verify-import-aliases.sh locally, the following
error happened:

  $ ./hack/verify-import-aliases.sh
  checking-imports:
  ERROR wrong alias for import "k8s.io/api/node/v1beta1" should be
  nodev1beta1 in file test/e2e/node/runtimeclass.go
   test/e2e_node/system/specs
  exit status 1

This fixes it.

k8s-ci-robot committed:
Merge pull request #84393 from wuyafang/remove_promhttp_from_kubelet

remove reference to promhttp in kubelet

k8s-ci-robot committed:
Merge pull request #84630 from SataQiu/clean-test-util-20191101

e2e: move KubectlVersion out of e2e test framework util.go

k8s-ci-robot committed:
Merge pull request #84806 from damemi/waitforstablecluster

Update WaitForStableCluster to wait for only system pods to exist

k8s-ci-robot committed:
Merge pull request #84871 from liggitt/cmd-test

Make test-cmd effective again

k8s-ci-robot committed:
Merge pull request #84905 from draveness/feature/remove-deprecated-priority-register

feat(scheduler): remove deprecated pattern in scheduler priority

yutedz committed:
Stop Watching when there is encoding error

Huang-Wei committed:
Update test logic to simulate NodeReady/False and NodeReady/Unknown events correctly

- optimize code to use one loop to spin up goroutines
- add `defer cleanupTest()` to avoid goroutine leaks
- use only one heartbeat channel

k8s-ci-robot committed:
Merge pull request #84510 from danielqsj/f-autoscaling

refactor autoscaling utils in e2e

k8s-ci-robot committed:
Merge pull request #84865 from deads2k/feature-gate-fail

panic in featuregate if a requested feature is unknown

k8s-ci-robot committed:
Merge pull request #84919 from RainbowMango/pr_functionality_for_customcollector_testing

provide functionality for custom collector testing

k8s-ci-robot committed:
Merge pull request #84937 from neolit123/1.17-fix-kubeadm-test-suite-flags

e2e_kubeadm: fix missing suite --test* flags

k8s-ci-robot committed:
Merge pull request #84941 from robscott/kubectl-get-e2e-fix

Limiting the scope of new kubectl get e2e tests to decrease flakiness

Commit:
fix: return NodeScore in even pods spread priority

k8s-ci-robot committed:
Merge pull request #84207 from robscott/endpointslice-hostname

Setting Hostname from Pods on EndpointSlice to match Endpoints behavior.

k8s-ci-robot committed:
Merge pull request #84693 from yutedz/watching-stop

Stop Watching when there is encoding error

k8s-ci-robot committed:
Merge pull request #84711 from chendotjs/kubelet-fixip

Refactor: use existing function to  simplify check for IPv6 in kubelet

k8s-ci-robot committed:
Merge pull request #84843 from clarklee92/UseFrameworkExpectIn-e2e

Use framework.ExpectEqual() in unit test

k8s-ci-robot committed:
Merge pull request #84864 from deads2k/optional-verify-opts

allow a verifyoptionsfunc to indicate that no certpool is available

ZP-AlwaysWin committed:
Remove unnecessary judgment

k8s-ci-robot committed:
Merge pull request #84868 from oomichi/move-e2e-util-2

Move functions from e2e/framework/util.go Part-2

k8s-ci-robot committed:
Merge pull request #84939 from alculquicondor/fix/go-mod

Set go version in fluentd-elasticsearch addon

RainbowMango committed:
Add NewLazyMetricWithTimestamp() API to stability framework.

RainbowMango committed:
Migrate custom collector and mark metrics with ALPHA.

liggitt committed:
Add testcases covering large valid patches

Commit:
clean up PriorityFunction

Signed-off-by: Zou Nengren <zouyee1989@gmail.com>

pdzwart-atlassian committed:
- Delete backing string set from a threadSafeMap index when the string set length reaches 0.

- Added thread_safe_store_test exercising new index backing string set delete at 0 functionality.

- TestThreadSafeStoreDeleteRemovesEmptySetsFromIndex logic nesting inverted.

- Added test case for usage of an index where post element delete there is non-zero count of elements and expect the set to still exist.

- Fixed date.

- Fixed awprice nits.

- Fix bazel.

k8s-ci-robot committed:
Merge pull request #81431 from andrewsykim/ga-node-topology

Promote Node Zone/Region Topology Labels to GA

k8s-ci-robot committed:
Merge pull request #84897 from YuikoTakada/delete-func-latencies

delete unused func in latencies.go

k8s-ci-robot committed:
Merge pull request #84911 from yue9944882/chore/bump-kube-openapi

Pin kube-openapi vendor to 30be4d16710a

k8s-ci-robot committed:
Merge pull request #84952 from draveness/fix/topology-spread-node-score

fix: return NodeScore in even pods spread priority

Commit:
feat(scheduler): convert InterPodAffinity to score plugin

k8s-ci-robot committed:
Merge pull request #84738 from liu-cong/service_affinity

Aggregate mulitple ServiceAffinity custom priorities to a single score plugin.

k8s-ci-robot committed:
Merge pull request #84942 from oomichi/fix-nodev1beta1

Fix the import aliase nodev1beta1

k8s-ci-robot committed:
Merge pull request #84896 from RainbowMango/pr_migrate_custom_collector_scheduler

Migrate custom collector for kube-scheduler

k8s-ci-robot committed:
Merge pull request #82698 from janario/fix/umount-subpath-warns

Unmount subpath should only scan the first level of files/directories

k8s-ci-robot committed:
Merge pull request #84299 from cwdsuzhou/Octo/add_policy_to_describe

Add reclaim policy to kubectl get storage class result

krzysied committed:
fixing docker fake client InspectExec

ahg-g committed:
Revert changes to WaitForStableCluster in scheduler e2e test

RainbowMango committed:
Deal with auto-generated files.
Update bazel by hack/update-bazel.sh

k8s-ci-robot committed:
Merge pull request #84773 from mm4tt/kubemark_cost_savings

Add envs for configuring hollow-node resource usage.

wojtek-t committed:
Correctly autogenerate conversions for autoscaling v2beta2

wojtek-t committed:
Autogenerated

k8s-ci-robot committed:
Merge pull request #78676 from i02sopop/kubectl_optimizations

Do some Kubectl optimizations suggested by the golangci linter

k8s-ci-robot committed:
Merge pull request #84006 from yutedz/k-ctrl-del-msg

Traverse resources before giving no name error

yutedz committed:
Move password retrieval to openstack_test.go

andrewsykim committed:
api/core/v1: add stable node instance-type label

Signed-off-by: Andrew Sy Kim <kiman@vmware.com>

andrewsykim committed:
kubelet: apply both beta and stable instance-type label

Signed-off-by: Andrew Sy Kim <kiman@vmware.com>

andrewsykim committed:
cloud node controller: apply stable instance-type label with backfill to existing nodes

Signed-off-by: Andrew Sy Kim <kiman@vmware.com>

andrewsykim committed:
scheduler: max attachable volume predicate should also check stable instance-type label

Signed-off-by: Andrew Sy Kim <kiman@vmware.com>

andrewsykim committed:
kubelet: add unit tests for backfilling stable os/arch/zone/region/instance-type labels

Signed-off-by: Andrew Sy Kim <kiman@vmware.com>

alculquicondor committed:
Remove leftover factory directory

Signed-off-by: Aldo Culquicondor <acondor@google.com>

andrewsykim committed:
noderestriction: update node restriction unit tests to use stable instance-type label

Signed-off-by: Andrew Sy Kim <kiman@vmware.com>

k8s-ci-robot committed:
Merge pull request #84351 from wojtek-t/promote_node_lease_to_GA

Promote node lease to GA

k8s-ci-robot committed:
Merge pull request #84523 from rajansandeep/skipmigration

Retain CoreDNS corefile when migration fails in kubeadm

k8s-ci-robot committed:
Merge pull request #84933 from krzysied/node_controller_v_level

Bumping up log level for not updated node

k8s-ci-robot committed:
Merge pull request #84983 from krzysied/docker_fake_client_fix3

Fixing docker fake client - InspectExec

k8s-ci-robot committed:
Merge pull request #84988 from ahg-g/ahg-stablecluster

Revert changes to WaitForStableCluster in scheduler e2e test

Commit:
Remove unused function from e2e framework util.go

The util.go file is so huge, it is better to make it small and simple
as possible for the maintenance.
This removes the following unused function:

- WaitForNodeHasTaintOrNot: Unused since 7e1794dcb1826ba8253064743ff021e85d69dfc3

Commit:
Move functions from e2e framework util.go

The following functions are called at some specific places only,
so this moves these functions to the places and makes them local.

- WaitForPersistentVolumeClaimDeleted: Moved to e2e storage
- PrintSummaries: Moved to e2e framework.go
- GetHostExternalAddress: Moved to e2e node
- WaitForMasters: Moved to e2e cloud gcp
- WaitForApiserverUp: Moved to e2e network
- WaitForKubeletUp: Moved to e2e storage vsphere

tallclair committed:
Add error detail

k8s-ci-robot committed:
Merge pull request #84963 from liggitt/feature-json-codes

Fix json patch limit check

k8s-ci-robot committed:
Merge pull request #84989 from wojtek-t/autogenerate_v2beta2_conversion

Autogenerate autoscaling/v2beta2 conversions

oomichi commented on test/e2e/framework/events/events.go:
```diff
@@ -35,11 +35,12 @@ import (
 	"github.com/onsi/ginkgo"
```
The dependencies seem clean and good for me for moving the package into e2e framework.

oomichi reviewed (commented):
Thanks for doing this.

/lgtm

Commit:
Enable verify-import-aliases check in CI

To block violation of import-aliases rule, this enables
verify-import-aliases in CI system.

Commit:
Remove potential Goroutine leak in kubeadm wait.go

There are two writes yet only one read on a non-buffered channel that is
created locally and not passed anywhere else.

Therefore, it could leak one of its two spawned Goroutines if either:
* The provided `f` takes longer than an erroneous result from
  `waiter.WaitForHealthyKubelet`, or;
* The provided `f` completes before an erroneous result from
  `waiter.WaitForHealthyKubelet`.

The fix is to add a one-element buffer so that the channel write happens
for the second Goroutine in these cases, allowing it to finish and freeing
references to the now-buffered channel, letting it to be GC'd.

k8s-ci-robot committed:
Merge pull request #81423 from codenrhoden/mount-testing-pkg

Refactor mount fakes

alculquicondor committed:
Add MetadataProducerFactory for predicates

Signed-off-by: Aldo Culquicondor <acondor@google.com>

k8s-ci-robot committed:
Merge pull request #82049 from andrewsykim/ga-node-instance-type-label

Promote Node Instance Type Label to GA

k8s-ci-robot committed:
Merge pull request #83385 from daxmc99/docstring-change

Fix typo in docstring of DeepEqual

k8s-ci-robot committed:
Merge pull request #84302 from serathius/restrict-prometheus-import

Restrict visibility of prometheus packages

k8s-ci-robot committed:
Merge pull request #84591 from TheLastProject/feature/print-join-as-admin-command

Let token create help with joining a master

k8s-ci-robot committed:
Merge pull request #84612 from soltysh/set_typesetter

Set TypeSetter in get print flags, like we do everywhere else

k8s-ci-robot committed:
Merge pull request #85002 from oomichi/cleanup-util

Remove unused function from e2e framework util.go

k8s-ci-robot committed:
Merge pull request #83840 from liggitt/json-iter

bump json-iterator dependency

neolit123 committed:
kubeadm: fix skipped etcd upgrade on secondary cp nodes

k8s-ci-robot committed:
Merge pull request #84331 from zouyee/cadprocess

Enable cAdvisor ProcessMetrics collecting

k8s-ci-robot committed:
Merge pull request #84562 from yutedz/drain-filter

Only put un-filtered pod in podDeleteList

k8s-ci-robot committed:
Merge pull request #84708 from yue9944882/feat/flowcontrol-storage

Feature: Flowcontrol rest storage implementation

k8s-ci-robot committed:
Merge pull request #84727 from danwinship/ipv6-bind

fix apiserver to advertise IPv6 endpoints if bound to IPv6

k8s-ci-robot committed:
Merge pull request #84883 from Huang-Wei/tbe-flake-followup

Update test logic to simulate NodeReady/False and NodeReady/Unknown events correctly

k8s-ci-robot committed:
Merge pull request #84917 from andyzhangx/deletedisk-check

fix race condition when delete azure disk right after that attach azure disk

k8s-ci-robot committed:
Merge pull request #84673 from hwdef/fix-staticcheck3

pkg/apis: fix staticcheck warning

ahg-g committed:
set default percentageOfNodesToScore in kube-scheduler back to 0 (maning adaptive)

k8s-ci-robot committed:
Merge pull request #84973 from draveness/feature/inter-pod-affinity-score

feat(scheduler): convert InterPodAffinity to score plugin

Commit:
fix kubectl diff panic

k8s-ci-robot committed:
Merge pull request #85024 from neolit123/1.17-fix-etcd-node-upgrade

kubeadm: fix skipped etcd upgrade on secondary CP nodes

hvaara committed:
Fix golint issues in pkg/apis/authorization

k8s-ci-robot committed:
Merge pull request #85030 from ahg-g/ahg-nodestoscore

Set default percentageOfNodesToScore in kube-scheduler back to "adaptive"

hvaara committed:
Fix golint issues in pkg/apis/policy

k8s-ci-robot committed:
Merge pull request #85009 from oomichi/enable-verify-import-aliases

Enable verify-import-aliases check in CI

neolit123 committed:
kubeadm: use the secure ports for kube-scheduler and kcm health checks

The insecure ports were deprecated in 1.12 and 1.13.

neolit123 committed:
kubeadm: remove the deprecated "--cri-socket" flag for "upgrade apply"

Remove the flag "--cri-socket" from the "kubeadm upgrade apply"
command. It has been marked as deprecated since 1.14.

k8s-ci-robot committed:
Merge pull request #85017 from LouisJackman/remove-goroutine-leak-in-kubeadm-wait

Remove potential Goroutine leak in kubeadm wait.go

hvaara committed:
Fix golint issues in pkg/apis/autoscaling/validation (#85041)

* Fix golint issues in pkg/apis/autoscaling/validation

* Fix spelling issue in field of Error

k8s-ci-robot committed:
Merge pull request #85008 from oomichi/move-utils-3

Move functions from e2e/framework/util.go Part-3

k8s-ci-robot committed:
Merge pull request #85037 from hvaara/fix-golint-pkg-apis-authorization

Fix golint issues in pkg/apis/authorization

k8s-ci-robot committed:
Merge pull request #85042 from hvaara/fix-golint-pkg-apis-policy

Fix golint issues in pkg/apis/policy

yutedz committed:
Handle error return from allocatePodResources

k8s-ci-robot committed:
Merge pull request #84992 from yutedz/openstack-passwd

Move password retrieval to openstack_test.go

ahmad-diaa committed:
make Configurator.Create return a Scheduler

k8s-ci-robot committed:
Merge pull request #84485 from tallclair/mirror-owner

Mirror owner

chiehting committed:
Fix golint issues in pkg/registry/core/service/storage

takanattie committed:
Fix a typo in pkg/controller

k8s-ci-robot committed:
Merge pull request #84648 from neolit123/1.17-fix-reset-hang

kubeadm: always use a short timeout for clientset creation

k8s-ci-robot committed:
Merge pull request #84948 from zouyee/PriorityFunction

clean up PriorityFunction

k8s-ci-robot committed:
Merge pull request #85043 from neolit123/1.17-enable-secure-ports

kubeadm: use the secure ports for kube-scheduler and kcm health checks

smarterclayton committed:
Always negotiate a decoder using ClientNegotiator

This commit performs two refactors and fixes a bug.

Refactor 1 changes the signature of Request to take a RESTClient, which
removes the extra copy of everything on RESTClient from Request. A pair
of optional constructors are added for testing. The major functional
change is that Request no longer has the shim HTTPClient interface and
so some test cases change slightly because we are now going through
http.Client code paths instead of direct to our test stubs.

Refactor 2 changes the signature of RESTClient to take a
ClientContentConfig instead of ContentConfig - the primary difference
being that ClientContentConfig uses ClientNegotiator instead of
NegotiatedSerializer and the old Serializers type. We also collapse
some redundancies (like the rate limiter can be created outside the
constructor).

The bug fix is to negotiate the streaming content type on a Watch()
like we do for requests. We stop caching the decoder and simply
resolve it on the request. We also clean up the dynamic client
and remove the extra WatchSpecificVersions() method by providing
a properly wrapped dynamic client.

k8s-ci-robot committed:
Merge pull request #83914 from guineveresaenger/ato-demo

Removes pkg/registry/core/secret/storage from golint failures

k8s-ci-robot committed:
Merge pull request #84243 from Dingshujie/fix-golint

Fix golint errors in pkg/apis/apps

k8s-ci-robot committed:
Merge pull request #83664 from RainbowMango/pr_refactor_kubelet_ut_with_metrics_testutil

Refactor kubelet ut with metrics testutil

k8s-ci-robot committed:
Merge pull request #84907 from RainbowMango/pr_migrate_custom_collector_kubelet

migrate kubelet custom metrics to stability framework part 1

k8s-ci-robot committed:
Merge pull request #84987 from RainbowMango/pr_migrate_custom_collector_kubelet_part2

migrate kubelet custom metrics to stability framework part 2

k8s-ci-robot committed:
Merge pull request #84135 from RainbowMango/pr_remove_backoff

Stop register to prom registry

k8s-ci-robot committed:
Merge pull request #84692 from smarterclayton/protocol_errors

Fix watch negotiation when using a non-default mime type in the client

k8s-ci-robot committed:
Merge pull request #82257 from prameshj/ilbsubnet

Support specifying a custom subnet for ILB ip in GCE

k8s-ci-robot committed:
Merge pull request #84796 from damemi/fix-graceful-shutdown

Fixed error check in graceful shutdown test

bertinatto committed:
Convert scheduler to use CSINode GA

nikhita committed:
publishing: bump Go versions in rules

deads2k committed:
close scheduler kube-apiserver

k8s-ci-robot committed:
Merge pull request #82572 from tnozicka/fix-rs-expectations

Fix RS expectations for recreate case

fabriziopandini committed:
alpha certs skip missing files

rosti committed:
kubeadm: Deprecate hyperkube use

As the hyperkube image is itself deprecated and moved out of tree, its use with
kubeadm gets deprecated too. Hence, deprecation messages will be printed when
it is used.

Signed-off-by: Rostislav M. Georgiev <rostislavg@vmware.com>

k8s-ci-robot committed:
Merge pull request #85090 from deads2k/schedulershutdown

close scheduler kube-apiserver

liggitt committed:
Plumb configured acceptContentType to client config

k8s-ci-robot committed:
Merge pull request #85094 from rosti/kubeadm-hyperkube-deprecation-notice

kubeadm: Deprecate hyperkube use

ahmad-diaa committed:
remove factory.Config struct and NewFromConfig

alejandrox1 committed:
Removed unused method from e2e test framework

This PR is in preparation to start cleaning up test/e2e/framework/pods.go.

Signed-off-by: alejandrox1 <alarcj137@gmail.com>

k8s-ci-robot committed:
Merge pull request #84996 from alculquicondor/cleanup/rm_factory

Remove leftover factory directory

yutedz committed:
Check that endpoint has subset before accessing first subset

davidz627 committed:
Remove plugin watching of deprecated directory {kubelet_root_dir}/plugins and support for CSI V0 in accordance with deprecation announcement in https://v1-13.docs.kubernetes.io/docs/setup/release/notes/

ravisantoshgudimetla committed:
BUILD files

k8s-ci-robot committed:
Merge pull request #84639 from SataQiu/clean-scheduler-20191101

scheduler: improve some comments and validation messages

k8s-ci-robot committed:
Merge pull request #85035 from gongguan/fix_diff_panic

fix kubectl diff panic

k8s-ci-robot committed:
Merge pull request #85044 from neolit123/1.17-deprecate-cri-socket-upgrade

kubeadm: remove the deprecated "--cri-socket" flag for "upgrade apply"

k8s-ci-robot committed:
Merge pull request #85067 from takanattie/fix_typo

Fix a typo in pkg/controller

alculquicondor committed:
Merging selectors for spreading into one

Signed-off-by: Aldo Culquicondor <acondor@google.com>

k8s-ci-robot committed:
Merge pull request #84524 from johnSchnake/customReporter

Print progress updates to stdout and publish to URL

k8s-ci-robot committed:
Merge pull request #84826 from yliaog/nodeselector

added nodeSelector: beta.kubernetes.io/os: linux

k8s-ci-robot committed:
Merge pull request #85012 from alculquicondor/feat/predicate_factory

Add MetadataProducerFactory for predicates

k8s-ci-robot committed:
Merge pull request #85092 from fabriziopandini/alpha-certs-skips-missing-certs

kubeadm: alpha certs should skip missing files

k8s-ci-robot committed:
Merge pull request #85095 from liggitt/protocol_errors

Plumb configured acceptContentType to client config

k8s-ci-robot committed:
Merge pull request #83815 from howardjohn/appprotocol

Add appProtocol to EndpointSlice.Port

k8s-ci-robot committed:
Merge pull request #80058 from xing-yang/snapshot_beta

Enable VolumeSnapshotDataSource Feature Gate and update e2e tests for VolumeSnapshot CRD v1beta1

k8s-ci-robot committed:
Merge pull request #83653 from cwdsuzhou/Octo/e2efix

Fix storage e2e clean up

k8s-ci-robot committed:
Merge pull request #82078 from SeongJuMoon/fix/storage-wrong-typo

fix wrong typo stoageMap to storageMap

k8s-ci-robot committed:
Merge pull request #80284 from danielqsj/exec

Fix a racing issue in client-go UpdateTransportConfig

k8s-ci-robot committed:
Merge pull request #83965 from robscott/endpointslice-managed-by

Adding new label to indicate what is managing an EndpointSlice

njuicsgz committed:
kube-proxy/ipvs get local addr just once per sync

k8s-ci-robot committed:
Merge pull request #84423 from mikedanese/tokbench

adjust token cache benchmarks to get more accurate behavior

k8s-ci-robot committed:
Merge pull request #84533 from davidz627/fix/deprecatedPath

 Remove plugin watching of deprecated directory and CSI v0 support in accordance with deprecation policy

k8s-ci-robot committed:
Merge pull request #84768 from liggitt/delegated-authnz-v1

switch delegated authnz to v1 APIs

k8s-ci-robot committed:
Merge pull request #84957 from ZP-AlwaysWin/dev-1105-1

Remove unnecessary judgment

k8s-ci-robot committed:
Merge pull request #85088 from nikhita/publishing-go-version

publishing: bump go versions in rules

okartau committed:
Removed clientset param from InjectContent

As one of previous changes added framework param to
InjectContent, it does not need separate param for clientset any more.

wojtek-t committed:
Eliminate couple unnecessary conversions

LukeShu committed:
client-go/tools: Docs: Clarify what's "old" core/v1 and what's "new" events/v1beta1

k8s-ci-robot committed:
Merge pull request #85059 from chiehting/registry/golint

Fix golint issues in pkg/registry/core/service/storage

pohly commented:
@danielqsj: can you rebase?

kkmsft committed:
Quick steps for generating hyperkube image

ahg-g committed:
Updated NewSnapshot interface to accept a NodeInfoMap instead of lists of nodes and pods

k8s-ci-robot committed:
Merge pull request #84672 from hwdef/fix-staticcheck2

cmd:fix staticcheck warning

k8s-ci-robot committed:
Merge pull request #84970 from atlassian/pdzwart/kubernetes/kubernetes/84959

- Delete backing string set from a threadSafeMap index when the string set length reaches 0.

k8s-ci-robot committed:
Merge pull request #85049 from ahmad-diaa/make-configurator-create-return-sched

Change Configurator.Create to Return a Scheduler

k8s-ci-robot committed:
Merge pull request #85134 from LukeShu/lukeshu/event-docs-non-api

client-go/tools: Docs: Clarify what's "old" core/v1 and what's "new" events/v1beta1

k8s-ci-robot committed:
Merge pull request #85135 from wojtek-t/delete_unnecessary_conversions_1

Eliminate couple unnecessary conversions

k8s-ci-robot committed:
Merge pull request #84813 from deads2k/admission-feature-gates

remove global variable dependency from admission plugins

k8s-ci-robot committed:
Merge pull request #84837 from justinsb/fix_get_kubelet_pods

e2e: Fix error where pods not logged

k8s-ci-robot committed:
Merge pull request #84862 from yutedz/master-subset-check

Check that endpoint has subset before accessing first subset

k8s-ci-robot committed:
Merge pull request #83568 from bertinatto/volume_limits_ga

Promote volume limits to GA

k8s-ci-robot committed:
Merge pull request #84179 from odinuge/startup-probe-test

Fix startup probe test by checking updated values

k8s-ci-robot committed:
Merge pull request #84718 from yastij/remove-validators

remove system validators package from kubeadm and use k8s.io/system-validators

k8s-ci-robot committed:
Merge pull request #84882 from marosset/run-as-username-beta

Moving WindowsRunAsUserName to beta

k8s-ci-robot committed:
Merge pull request #84924 from njuicsgz/master

proxy/ipvs: only get local addresses once per sync

k8s-ci-robot committed:
Merge pull request #85047 from yutedz/dev-mgr-err-handling

Handle error return from allocatePodResources

k8s-ci-robot committed:
Merge pull request #85100 from alejandrox1/cleanup-e2epod

Removed unused method from e2e test framework

k8s-ci-robot committed:
Merge pull request #85106 from alculquicondor/refactor/single_selector

Merging selectors used for spreading into one

k8s-ci-robot committed:
Merge pull request #85120 from kkmsft/quick_start_hyperkube

Quick steps for generating hyperkube image

k8s-ci-robot committed:
Merge pull request #85139 from ahg-g/ahg-snapshot2

Updated NewSnapshot interface to accept a NodeInfoMap

k8s-ci-robot committed:
Merge pull request #84522 from liu-cong/plugin

Add scheduler plugin execution duration metric.

k8s-ci-robot committed:
Merge pull request #79223 from rosti/kubeadm-component-configs-versioned

kubeadm: component configs versioned

JishanXing committed:
76586 Add generated license files

k8s-ci-robot committed:
Merge pull request #76310 from ravisantoshgudimetla/fix-priority-quota

Relax namespace restriction for critical pods

k8s-ci-robot committed:
Merge pull request #84042 from okartau/avoid-local-kubectl

test/e2e: Reduce need to use local kubectl

k8s-ci-robot committed:
Merge pull request #76586 from JishanXing/76093

Fix #76093, restructure LICENSES file generation

danielqsj commented:
/close

danielqsj commented:
@pohly rebase has some issues. I open another PR for this fix https://github.com/kubernetes/kubernetes/pull/85190.
